src/index.jsx

```diff
@@ -1,17 +1,58 @@
-import React from 'react';
+import React,{ useEffect, useState } from 'react';
 import { render } from 'react-dom';
+
+// styles
 import './style.css';
 
+// components
+import AnimalDetail from './components/AnimalDetail/AnimalDetail';
+import AnimalList from './components/AnimalList/AnimalList';
+
 const App = () => {
+  
+  const [listOfAnimals, setListOfAnimals] = useState([]);
+  const [chosenAnimal, setChosenAnimal] = useState([]);
```

Comment:
ve stavu se typicky nemixují datové typy (vede to k problémům). Tady jako "prázdný" typ máš pole, kdežto pak tam dáš objekt. Použil bych jako defaultní hodnotu buď prázdný objekt (`{}`) a nebo ještě lépe undefined/null protože to zvíře v tu chvíli prostě není, tj logicky legitimně tam v tuhle chvíli je "nic"